super-ddf.c

```diff
@@ -5195,6 +5195,19 @@ static void default_geometry_ddf(struct supertype *st, int *level, int *layout,
 		*layout = ddf_level_to_layout(*level);
 }
 
+static int update_super_ddf_dummy(struct supertype *st, struct mdinfo *info,
+			 enum update_opt update,
+			 char *devname, int verbose,
+			 int uuid_set, char *homehost)
+{
+	/*
+	 * A dummy update_super function is needed
```

Comment:
Thank you for addressing that. I have added the reason in the commit message and found a related discussion about this in the maillist.


```
commit e3d493d7046cc07b4eca95a2cc24cc59e0ee3b34
Author: lilinzhe <llz@antiy.cn>
Date:   Mon Dec 16 12:00:02 2024 +0800

    super-ddf: Prevent crash when handling DDF metadata
    
    A dummy function is defined because availability of ss->update_super is
    not always verified.
    
    This fix addresses a crash reported when assembling a RAID array using
    mdadm with DDF metadata. For more details, see the discussion at:
    https://lore.kernel.org/all/
    CALHdMH30LuxR4tz9jP2ykDaDJtZ3P7L3LrZ+9e4Fq=Q6NwSM=Q@mail.gmail.com/
    
    The discussion centers on an issue with mdadm where attempting to
    assemble a RAID array caused a null pointer dereference. The problem
    was traced to a missing update_super() function in super-ddf.c, which
    led to a crash in Assemble.c.
    
    Signed-off-by: lilinzhe <llz@antiy.cn>
```